### PR TITLE
WorkFlow engine improve error handling

### PR DESCRIPTION
**Pull request by fcaffieri**
# Description

The objective of this PR is:
- Improve error handling to avoid failures that do not represent the real problem.
- Improve module logging.
- Improve validations
- Capture the cancellation by the user.
---

## Testing performed

Interrupt by user:

image

Normal execution:

image

### CODE REVIEW

**Review by pro-akim — CHANGES_REQUESTED** [review]
## Review Notes

Some comments were done

File: deployability/modules/workflow_engine/task.py [hunk 1]
```diff
@@ -56,9 +64,12 @@ def execute(self) -> None:
             )
             logger.debug(f'Finished task "{self.task_name}" execution with result:\n{str(result.stdout)}')
 
-            if result.returncode != 0:
-                raise subprocess.CalledProcessError(returncode=result.returncode, cmd=result.args, output=result.stdout)
+            #if result.returncode != 0:
```
[the remark this answers is not in the document]
Comment (fcaffieri) [reply deployability/modules/workflow_engine/task.py]: Done in [d865900](https://github.com/wazuh/wazuh-qa/pull/4960/commits/d8659006955a6b65f83e9cf30788978486c19695)

**Review by pro-akim — APPROVED** [review]
## Review Notes
LGTM

File: deployability/modules/workflow_engine/__main__.py [hunk 2]
```diff
@@ -1,10 +1,12 @@
+
```
Comment (rauldpm) [inline deployability/modules/workflow_engine/__main__.py:1]: Remove this empty line

File: deployability/modules/workflow_engine/task.py [hunk 3]
```diff
@@ -45,7 +49,11 @@ def execute(self) -> None:
                     task_args.extend([f"--{key}={argvalue}" for argvalue in value])
                 else:
                     task_args.append(f"--{key}={value}")
-        self.logger.debug(f'Running task "{self.task_name}" with arguments: {task_args}')
+            else:
+                logger.error(f'Coulg not parse arguments {arg}')
```
Comment (rauldpm) [inline deployability/modules/workflow_engine/task.py:53]: Typo: Coul`g` -> Coul`d`

File: deployability/modules/workflow_engine/task.py [hunk 4]
```diff
@@ -59,6 +67,9 @@ def execute(self) -> None:
             if result.returncode != 0:
                 raise subprocess.CalledProcessError(returncode=result.returncode, cmd=result.args, output=result.stdout)
         except subprocess.CalledProcessError as e:
+            error_msg = e.stderr
+            if "KeyboardInterrupt" in error_msg:
+                raise KeyboardInterrupt(f"Error executing process task with keyboard interrupt")
```
Comment (rauldpm) [inline deployability/modules/workflow_engine/task.py:72]: We need to set some style criteria, in this case, a sentence ended with a dot or not

File: deployability/modules/workflow_engine/workflow_processor.py [hunk 5]
```diff
@@ -33,9 +34,14 @@ def __validate_schema(self, workflow_file: Path | str) -> None:
         Args:
             workflow_file (Path | str): Path to the workflow file.
         """
-        validator = SchemaValidator(self.schema_path, workflow_file)
-        validator.preprocess_data()
-        validator.validateSchema()
+        try:
+            logger.debug(f"Validating inptu file: {workflow_file}")
```
Comment (rauldpm) [inline deployability/modules/workflow_engine/workflow_processor.py:38]: Typo: inp`tu` -> inp`ut`

**Review by rauldpm — CHANGES_REQUESTED** [review]
GJ, some style errors

File: deployability/modules/workflow_engine/workflow_processor.py (hunk 5 above)
Comment (fcaffieri) [reply deployability/modules/workflow_engine/workflow_processor.py]: Done in [ebce158](https://github.com/wazuh/wazuh-qa/pull/4960/commits/ebce158066ecff2dc0cb82cc7f902dc54b6d489d)

File: deployability/modules/workflow_engine/task.py (hunk 4 above)
Comment (fcaffieri) [reply deployability/modules/workflow_engine/task.py]: Done in [ebce158](https://github.com/wazuh/wazuh-qa/pull/4960/commits/ebce158066ecff2dc0cb82cc7f902dc54b6d489d)

File: deployability/modules/workflow_engine/task.py (hunk 3 above)
Comment (fcaffieri) [reply deployability/modules/workflow_engine/task.py]: Done in [ebce158](https://github.com/wazuh/wazuh-qa/pull/4960/commits/ebce158066ecff2dc0cb82cc7f902dc54b6d489d)

File: deployability/modules/workflow_engine/__main__.py (hunk 2 above)
Comment (fcaffieri) [reply deployability/modules/workflow_engine/__main__.py]: Done in [ebce158](https://github.com/wazuh/wazuh-qa/pull/4960/commits/ebce158066ecff2dc0cb82cc7f902dc54b6d489d)

File: deployability/modules/allocation/aws/instance.py [hunk 6]
```diff
@@ -68,9 +68,9 @@ def ssh_connection_info(self) -> ConnectionInfo:
             ConnectionInfo: SSH connection information.
         """
         return ConnectionInfo(hostname=self._instance.public_dns_name,
-                              user=self._user,
-                              port=22,
-                              private_key=str(self.credentials.key_path))
+                                user=self._user,
+                                port=22,
+                                private_key=str(self.credentials.key_path))
```
[the remark this answers is not in the document]
Comment (fcaffieri) [reply deployability/modules/allocation/aws/instance.py]: Done in [ebce158](https://github.com/wazuh/wazuh-qa/pull/4960/commits/ebce158066ecff2dc0cb82cc7f902dc54b6d489d)

File: deployability/modules/allocation/allocation.py [hunk 7]
```diff
@@ -64,7 +64,7 @@ def __delete(cls, payload: models.DeletionPayload) -> None:
 
         Args:
             payload (DeletionPayload): The payload containing the parameters
-                                              for instance deletion.
+                                                for instance deletion.
```
[the remark this answers is not in the document]
Comment (fcaffieri) [reply deployability/modules/allocation/allocation.py]: Done in [ebce158](https://github.com/wazuh/wazuh-qa/pull/4960/commits/ebce158066ecff2dc0cb82cc7f902dc54b6d489d)

**Review by rauldpm — APPROVED** [review]

**Pull request merged**